src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -78,14 +78,6 @@ class BaseOptionsSelector extends Component {
             }
         }
 
-        if (this.textInput && this.props.autoFocus && !prevProps.isFocused && this.props.isFocused) {
-            InteractionManager.runAfterInteractions(() => {
-                // If we automatically focus on a text input when mounting a component,
-                // let's automatically focus on it when the component updates as well (eg, when navigating back from a page)
-                this.textInput.focus();
-            });
-        }
-
```

Comment:
We cannot simply remove this code. This breaks auto focus in android.

Reply:
We might consider using `focusWithDelay.js`. Didn't investigate it deeply, just a quick thought

Reply:
`focusWithDelay.js` is for composer only.

Let's follow this pattern Rory [suggested](https://expensify.slack.com/archives/C01GTK53T8Q/p1694660990479979)

```
const focusTimeoutRef = useRef(null);
useFocusEffect(useCallback(() => {
    focusTimeoutRef.current = setTimeout(textInputRef.current.focus, CONST.ANIMATED_TRANSITION);
    return () => {
        if (!focusTimeoutRef.current) {
            return;
        }
        clearTimeout(focusTimeoutRef.current);
    };
}, []));
```